src/saturn_engine/worker_manager/config/declarative_job_definition.py

```diff
@@ -0,0 +1,93 @@
+import dataclasses
+import functools
+
+import desert
+import marshmallow
+
+from saturn_engine.core import api
+
+from .declarative_base import BaseObject
+from .declarative_pipeline import PipelineSpec
+from .static_definitions import StaticDefinitions
+
+
+@dataclasses.dataclass
+class JobInput:
+    inventory: str
+
+    def to_core_object(
+        self,
+        static_definitions: StaticDefinitions,
+    ) -> api.InventoryItem:
+        return static_definitions.inventories[self.inventory]
+
+
+@dataclasses.dataclass
+class JobOutput:
+    topic: str
+
+    def to_core_object(
+        self,
+        static_definitions: StaticDefinitions,
+    ) -> api.TopicItem:
+        return static_definitions.topics[self.topic]
+
+
+@dataclasses.dataclass
+class JobDefinitionTemplatePipeline:
+    name: str
+    args: dict[str, object]
+    resources: dict[str, str]
+
+
+@dataclasses.dataclass
+class JobDefinitionTemplate:
+    name: str
+    input: JobInput
+    output: dict[str, list[JobOutput]]
+    pipeline: PipelineSpec
+
+    def to_core_object(
+        self,
+        static_definitions: StaticDefinitions,
+    ) -> api.QueueItem:
+        return api.QueueItem(
+            name=self.name,
+            input=self.input.to_core_object(static_definitions),
+            output={
+                key: [t.to_core_object(static_definitions) for t in topics]
+                for key, topics in self.output.items()
+            },
+            pipeline=api.QueuePipeline(
```

Comment:
shouldn't we change the template to match the object?

```
class PipelineInfo:
  name: ...
  resources: ...

class PipelineSpec:
  info: PipelineInfo
  args: ...
```

Reply:
Sure. Please post full config examples and/or edit the gitlab issue if you want to propose changes, its hard to guess from the incomplete small bribes.